features/achordion.c

```diff
@@ -116,6 +125,9 @@ bool process_achordion(uint16_t keycode, keyrecord_t* record) {
       }
     }
 
+#if ACHORDION_TYPING_STREAK_TIMEOUT > 0
+    idle_timer = timer_read();
```

Comment:
Yes that's my understanding. I tried different alternative, and this implementation was the one that worked the best. Resetting the timer on tap-hold key press during `STATE_RELEASED` resulted in false negative (tap instead of hold).

Though I'm not entirely sure I fully understand the interaction between process_record and the recursive call from Achordion, so this might need some adjustments.

Reply:
Thanks! That all looks good to me.